lib/PGalias.pm

```diff
@@ -205,41 +206,36 @@ sub make_alias {
 	# dispatch	
 #######################################################################
 	# determine extension, if there is one
-	# if extension exists, strip and use the value for $ext
+	# if extension exists use the value for $ext
 	# files without extensions are flagged with errors.
-
-	#      								$self->debug_message("This auxiliary file id is $aux_file_id" );
-	if ($aux_file_id =~ s/\.([^\.]+)$// ) {
+	# The extension is retained as part of  aux_file_id 
+	
+	#$self->debug_message("This auxiliary file id is $aux_file_id" );
+	if ($aux_file_id =~ m/\.([^\.]+)$/ ) {
 		$ext = $1;
 	} else {
 		$self->warning_message( "The file name $aux_file_id did not have an extension.<BR> " .
 		     "Every file name used as an argument to alias must have an extension.<BR> " .
-		     "The permissable extensions are .jpg, .pdf, .gif, .png, and .html .<BR>");
+		     "The permissable extensions are .jpg, .pdf, .gif, .png, .mpg, .mp4, .ogg, .webm and .html .<BR>");
 		$ext  = undef;
```

Comment:
Its fine with me as long as my stuff gets tested.  Also, after thinking
about it adding a `video` command might not be a bad idea.  (See my other
pull)


On Sun, Jan 3, 2016 at 5:51 PM, Michael Gage <notifications@github.com>
wrote:

> In lib/PGalias.pm
> <https://github.com/openwebwork/pg/pull/255#discussion_r48698354>:
>
> >  		$ext = $1;
> >  	} else {
> >  		$self->warning_message( "The file name $aux_file_id did not have an extension.<BR> " .
> >  		     "Every file name used as an argument to alias must have an extension.<BR> " .
> > -		     "The permissable extensions are .jpg, .pdf, .gif, .png, and .html .<BR>");
> > +		     "The permissable extensions are .jpg, .pdf, .gif, .png, .mpg, .mp4, .ogg, .webm and .html .<BR>");
> >  		$ext  = undef;
>
> Is that ok with you? It seemed useful to just combine the two pulls. As it
> turns out I didn't do any more refactoring (no time) but eventually I want
> to look at this file again and see if the html and hardcopy paths can be
> refactored. They still share a lot of code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openwebwork/pg/pull/255/files#r48698354>.
>